### PR TITLE
backend/feat : added delete api for person from dashboard

### PR DESCRIPTION
**Pull request by ashishsaini25**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

added delete api for person from dashboard
### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/Lib/src/Domain/Action/Dashboard/Person.hs [hunk 1]
```diff
@@ -418,6 +418,22 @@ changeEmailByAdmin _ personId req = do
...
+deletePerson ::
+  ( BeamFlow m r,
+    Redis.HedisFlow m r,
+    HasFlowEnv m r '["authTokenCacheKeyPrefix" ::: Text]
+  ) =>
+  TokenInfo ->
+  Id DP.Person ->
+  m APISuccess
+deletePerson _ personId = do
+  void $ QP.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
```
Comment (0utkarsh) [inline Backend/app/dashboard/Lib/src/Domain/Action/Dashboard/Person.hs:430]: We can use replica

**Review by 0utkarsh — APPROVED** [review]
Approving as code wise everything is correct. Can we implement soft delete in these cases(when we are not forced to delete according to play store compliance etc in case of drivers/customers)?

**Comment by ashishsaini25** [comment]
> Approving as code wise everything is correct. Can we implement soft delete in these cases(when we are not forced to delete according to play store compliance etc in case of drivers/customers)?

yes , i am thinking to add enable and disable functionality.

**Comment by ashishsaini25** [comment]
> > yes , i am thinking to add enable and disable functionality.
> 
> Sure, this can be used if enable-disable will go back and forth. If we want to soft delete, we can just add a column like `deleted_at` and while listing users or performing anything, we can keep a check on this field being not null

sure that can also be the way , but currently i have added enabled in registration token where it will handle this case.

**Review by vijaygupta18 — APPROVED** [review]

**Pull request merged**